utility/src/main/java/org/oppia/util/profile/DirectoryManagementUtil.kt

```diff
@@ -0,0 +1,21 @@
+package org.oppia.util.profile
+
+import javax.inject.Inject
+import javax.inject.Singleton
+import org.oppia.app.model.ProfileId
+import java.io.File
+
+/** Utility to manage creation and deletion of directories */
+@Singleton
+class DirectoryManagementUtil @Inject constructor(){
+
+  /** Get or creates a directory with the name specified by profileId */
+  fun getOrCreateDir(profileId: ProfileId): File {
```

Comment:
To avoid the dependence, maybe just take in a string for the directory name and upstream only use the profile ID as the directory name. I'd expect only the profile system to depend on this utility, anyway.